pkg/knuu/instance_old.go

```diff
@@ -0,0 +1,342 @@
+// Package knuu provides the core functionality of knuu.
+package knuu
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/celestiaorg/knuu/pkg/builder"
+	"github.com/celestiaorg/knuu/pkg/instance"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type Instance struct {
+	instance.Instance
+}
+
+type Executor struct {
+	*Instance
+}
+
+type InstancePool struct {
+	instance.InstancePool
+}
+
+type InstanceState instance.InstanceState
+
+const (
+	None InstanceState = iota
+	Preparing
+	Committed
+	Started
+	Stopped
+	Destroyed
+)
+
+func NewInstance(name string) (*Instance, error) {
```

Comment:
Could this be `func (k *Knuu) NewInstance(name string) (*Instance, error)`?

Then, we could keep a list of all instances created in that scope.

If we have this, we can have something like `knuu.Destroy()` that deletes all instances in that scope.

Reply:
Wouldn't that change need to be added here: https://github.com/celestiaorg/knuu/pull/356/files#diff-3ef3ea3ddb9b6f3d6f910f9da360132513c08de27816ea4956ce5f28d1b08aa2R11

Since this `old` file is for backwards compatibility and shouldn't be changed? 

I would also recommend the convention of `func (k *Knuu) AddInstance()` if the intention of the method is to add an instance to the knuu object.

Reply:
right now we have `knuu.Cleanup()` which we can rename it to `knuu.Destroy()` that deletes the entire namespace and so for that particular case we do not need to keep track of instance objects somewhere in knuu by using `func (k *Knuu) AddInstance()`. 
@smuu @MSevey what other usecases do you think we might want to have by keeping references to instance objects in the knuu object?

Reply:
defer to sam on this one. 
Also we can probably just continue this discussion in a follow up.

Reply:
With the batch processing that we are going to work on, this might come handy.

Reply:
@smuu any suggestions?